application/comit_node/src/http_api/swap_resource.rs

```diff
@@ -22,6 +22,7 @@ use serde::Serialize;
 )]
 pub struct SwapResource<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset, S: Serialize> {
     pub role: String,
+    pub counterparty: String,
```

Comment:
This could be `Http<PeerId>` instead of just a string. You will have to add an `impl Serialize for Http<PeerId>` but that makes sense anyway.